### DISCUSSION TITLE
'this' undefined in Typescript project

### DISCUSSION THREAD

**Discussion by RyannGalea** [swc-project/swc#2205] [Q&A]
@kdy1 

If you remember I was having issues with my Graphql project renaming classes, I have since upgraded to the latest version & it seems to have solved the renaming issue and I am able to get the project to compile and run. But if I change my settings to below I get the same issue arises.

```
  "env": {
    "targets": {
      "node": "14"
    }
  }
```

When I run the app without that above it works but I get issues with my decorator functions not being able to access 'this' where previously I was able to. Not sure if my settings are wrong. or I'm doing something else wrong. 

```

@ObjectType()
@pre<Booking>(['find', 'findOne'], async function() {
  console.log('this', this)    <=== undefined
  const find = this['_conditions']
  await BookingModel.updateMany(find, [{'$set': {'bookingStatus': GetStatusConditions()}}]).exec()
})
export class Booking extends Universal  {

  @Field(type => String)
  @prop({type: String, index: true})
  autoReference: string
 
  @Field(type => Property)
  @prop({ref: EntitiesEnum.PROPERTY})
  property: Ref<Property>

}

export const BookingModel = getModelForClass(Booking)

```

Any ideas 7 is the node target an issue ?

```
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false,
      "decorators": true,
      "dynamicImport": true
    },
    "target": "es5",
    "loose": true,
    "transform": {
      "legacyDecorator": true,
      "decoratorMetadata": true
    }
  },
  "module": {
    "type": "commonjs"
  }
} 
```

**Comment by kdy1** [comment]
Hmm... This seems like a bug in a pass that is enabled only for such config.
Can you file an issue?
